### ISSUE TITLE
Load subscribers from the twitch API to allow subscribers to lurk

### ISSUE THREAD

**Issue by liquidnya** [fluid-queue/fluid-queue#105]
**Is your feature request related to a problem? Please describe.**
Whenever the stream is started and someone is subscribed and they just want to lurk, then they won't get their subscriber multiplier for their weight, unless they put a message into chat. They also wouldn't get picked by `!subnext` or `!subrandom`.

**Describe the solution you'd like**
When the bot starts up it could get a list of all subscribers.

**Describe alternatives you've considered**
The list of all subscribers could be fetched for every `waitingTimerTick`, but then cached like the online chatters, but for a longer time, for example subscribers could be cached for 5 hours (which makes it less than half the time active chatters are kept as subscribers).

We could also listen for subscriber events (after loading subscribers at startup).

**Additional context**
To get a list of subscribers (<https://dev.twitch.tv/docs/api/reference/#get-broadcaster-subscriptions>) a token from the brodcaster is needed.

Either way another token file is needed for this optional feature (I would suggest making it an optional feature because of this).
And then that token file would need to be loaded and setup correctly in `src/twitch-api.ts`.

**Comment by demize** [comment]
> Either way another token file is needed

No, I think this could be done with the one token, we'd just need to add `channel:read:subscriptions` to the scopes we request. It would still be a breaking change (since we don't currently request that scope) but it's not quite as bad as adding a whole second token.

**Comment by demize** [comment]
We probably also don't need to make it a setting... or at least not one we expose in settings.json. twurple keeps track of the scopes the token has (accessible through `this.#authProvider?.getCurrentScopesForUser(this.#botUserId);` at least, probably better ways) so we can probably just take that list of scopes after the bot connects to IRC and use that to gate features like this. Probably print a warning when they start the bot that it's missing this scope too, since this functionality is probably pretty uncontroversial.

**Comment by demize** [comment]
Made some progress on this... the main issue I'm having now is building a chatter to put into `subscribers`. And that's mostly because it insists on having an `equals` function...

I'll worry about this more tomorrow.

**Comment by liquidnya** [comment]
@demize There is a function you can probably re-use: `isQueueSubmitter` like this:

https://github.com/fluid-queue/fluid-queue/blob/0dd51a116f15fd73196ef00497aab005d7244cb8/src/chatbot.ts#L22-L24

**Comment by demize** [comment]
Ah, good idea. And maybe eventually supplement that with pub-sub for new subscribers during the stream (which I do not want to think about until I've got this working).

Probably also check for mods, since they can lurk too. Easy enough to do at the same time (though I don't know about pub-sub to monitor for new mods).

**Comment by liquidnya** [comment]
One thing we could do with moderators, is remove moderators if they are banned or timed out:
(using `tmi.js`)
```typescript
client.on("timeout", (_0, _1, _2, _3, userState) => {
  userState["target-user-id"];
});
client.on("ban", (_0, _1, _2, userState) => {
  userState["target-user-id"];
});
```
But I think we can not listen to `/unmod` events through `tmi.js` as there is no IRC message being sent when that happens.

I am not sure if it matters that much, as the moderator cache is only used for level selection and checking the moderator status for chat commands is always done by checking the moderator chat badge anyways.

**Comment by demize** [comment]
Merged as part of 2.0.0-rc.1.
